### DISCUSSION TITLE
Accuracy

### DISCUSSION THREAD

**Discussion by JordanEmil** [Part-II-Computational-Physics/.github#75] [General]
I'm getting a bunch of true/false outputs when printing the accuracy... 

`def loss(Xdat,ydat,model):
    """Returns the total loss (C(theta) + regularisation) and the accuracy of the neural network."""
    
    inputs = [list(map(Value, xrow)) for xrow in Xdat]
    scores = list(map(model, inputs))
    
   
    # Compute the SVM max-margin loss    
    losses = [(1 + -yi*scorei).relu() for yi, scorei in zip(ydat, scores)]
    data_loss = sum(losses) * (1.0 / len(losses))
   
    
    
    alpha = 1e-4
    reg_loss = alpha * sum((p*p for p in model.parameters()))

    total_loss = data_loss + reg_loss #loss function + regularisation

    accuracy = [(yi > 0) == (scorei.data > 0) for yi, scorei in zip(ydat, scores)]
    
    return total_loss, accuracy

total_loss, acc = loss(X, y, model)
print(total_loss, acc)`

my output is:

> Value(data=1.261231290849708, grad=0) [False, True, True, True, True, True, False, False, True, True, True, False, False, True, False, True, True, False, True, True, True, False, True, True, False, True, True, True, True, False, False, True, True, False, False, True, False, False, False, True, True, False, False, False, False, False, False, False, True, True, False, True, False, True, True, True, True, False, True, False, False, True, True, False, False, False, True, False, True, True, False, True, False, True, False, False, False, True, False, False, False, False, True, False, False, True, False, True, False, True, False, True, True, False, True, True, False, False, True, False]

Just looking at the micrograd/demo.ipynb I should be getting 0.5 for the accuracy (which is evident from the true/false ratio). The problem is this then affects Task 2. 

`if step % 10 == 0 or step == n_steps-1:
            print(f"step {step} loss {total_loss.data}, accuracy {acc*100}%")`

It cannot give me %s

Any help would be incredible, thank you.

**Comment by JordanEmil** [comment]
Should the code:

`
return total_loss, sum(accuracy) / len(accuracy)
`

?

I am hesitant to change any code you guys have put in there.

**Reply by tony-teng-yt377** [reply]
Yeah I had the same issue, and we just need to change it that way

**Comment by ceciliegl** [comment]
Yes, the code should return sum(accuracy) / len(accuracy)!
